include/openthread/channel_manager.h

```diff
@@ -47,21 +47,28 @@ extern "C" {
  * @brief
  *   This module includes functions for Channel Manager.
  *
- *   The functions in this module are available when Channel Manager feature
- *   (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE`) is enabled. Channel Manager is available only on an FTD build.
+ *   The functions in this module are available when Channel Manager features
+ *   (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE`) and (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE`) is
+ *   enabled. Channel Manager behavior depends on the device role. On a Full Thread Device (FTD) manages the
+ *   network-wide Pan channel, or with (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE`) set, selects Csl
+ *   channel if FTD is downgraded to a SSED role. On a Minimal Thread Device
+ *   (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE`) selects the local Csl channel.
  *
  * @{
  *
  */
 
 /**
- * Requests a Thread network channel change.
+ * Requests a Thread network channel, or Csl channel change.
  *
  * The network switches to the given channel after a specified delay (see #otChannelManagerSetDelay()). The channel
- * change is performed by updating the Pending Operational Dataset.
+ * change is performed by updating the Pending Operational Dataset on a Full Thread Device.
  *
  * A subsequent call will cancel an ongoing previously requested channel change.
  *
+ * With (`OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE`) set on a Csl synchronized device, the Csl
+ * channel is changed instead of the network channel.
+ *
```

Comment:
Regarding CSL change, the change is always local to this device. The question for me is if we even need to update this API to allow it to be used for CSL change since we already have `otLinkSetCslChannel()` API which allows user to change the CSL channel.

"Channel Manager" provides two main functions:
- It adds a new API `otChannelManagerRequestChannelChange()` for user to request a channel change of whole Thread Network
   - For CSL we already `otLinkSetCslChannel()` for this
- It adds `AutoChannelSelection` function where the manager itself decide to switch based on data collected by `ChannelMonitor`
  - This can be useful for CSL, and to make it explicit we can consider adding it as its own API/feature `otChannelManagerSetAutoCslChannelSelectionEnabled()` which enable automatic CSL channel selection.

Thoughts?

Reply:
I think we can distinguish between FTD and MTD builds. For later it is barely relevant, current proposal allows using same API to start and configure channel manager on both builds.

FTD downgrading to Csl links will not occur automatically to my understanding. It is explicit choice of the user, similar for upgrading back to REED or router role. Therefore, user can also reconfigure channel manager for those roles as desired with existing API. Eg. if user would want to manage Csl channel as SSED on downgraded FTD, but not network channel as router, this can be done with existing API, but otChannelManagerSetAutoCslChannelSelectionEnabled() would simplify things (just for this case).

Should roles and link types in future change autonomously, I would still expect ChannelManager activated on FTD selecting a channel also when downgraded to SSED. In particular, as the channel selection is always a local matter based on ChannelMonitor and for SSED an instant local decision changing to the selected alternative channel.

Analog to `otLinkSetCslChannel()` also `otLinkSetChannel()` is available.

Thoughts?

Reply:
There are some key differences between `otLinkSetChannel` and `SetCslChannel`.
- `oLinkSetChannel` can only be used when Thread is disabled and only set the channel on this device alone. It cannot be used to change the channle on an already attached device and/or on entire network. This API is primary intended for testing.
```c++
**
 * Set the IEEE 802.15.4 channel
 *
 * Succeeds only when Thread protocols are disabled.  A successful call to this function invalidates the
 * Active and Pending Operational Datasets in non-volatile memory.
 *
 ```
- `otLinkSetCslChannel()`  on the other hand, can be used at anytime (when attached/detached) and will take care of informing parent of newly set CSL channel.

Reply:
I think network channel selection and CSL channel selection are basically two different things and shouldn't be done in one API. (Even we add enough comments, I think it's kind of confusing and make the code more complex)

I feel that currently they are put in the same API because the channel monitoring are the same for them and it can be reused. My suggestion is to not using the same API for the two things and let it have different behaviors under different roles. I prefer: if a user wants auto network channel selection behavior changes to auto csl channel selection behavior, the user should explicitly call the API again. 

It may be hard to say which one is better in an objective way. Different users may have different preference. As far as I'm concerned, I feel that network channel selection and CSL channel selection are two different things so we'd better make the API calls explicit if a user wants to use CSL channel selection behavior. Thoughts?

Reply:
Thx for your helpful explanations. I remove the capability to change CSL channel from `ChannelManager` API. Consequently the API for Cli commands which also trigger a channel change, e.g. `channel manager change <channel>` and `channel manager select` become non-functional for FTD downgraded to csl link and not be available on MTD, while internally the code is still needed for the auto selection case on FTD builds.

Also I add `otChannelManagerSetAutoCslChannelSelectionEnabled` and `otChannelManagerGetAutoCslChannelSelectionEnabled` in addition. Thus on FTD downgraded to csl link both `otChannelManagerSetAutoChannelSelectionEnabled` and `otChannelManagerSetAuto`CslChannelSelectionEnabled` might be set for selecting network and csl channel, respectively. In this case, only the channel of the active link will be managed.

Let me know what you think.

Reply:
SGTM. Thanks 👍

Reply:
SGTM too. Thanks.

Reply:
Updated.